src/Operation/Factory/SubresourceOperationFactory.php

```diff
@@ -79,6 +80,15 @@ private function computeSubresourceOperations(string $resourceClass, array &$tre
             $subresourceMetadata = $this->resourceMetadataFactory->create($subresourceClass);
 
             $visiting = "$resourceClass $property $subresourceClass";
+
+            // Handle maxDepth
+            if ($rootResourceClass == $resourceClass) {
+                $maxDepth = $subresource->getMaxDepth();
+            }
+
+            if(!is_null($maxDepth) && count($visited) >= $maxDepth) {
```

Comment:
We will try to avoid `is_null` and simply use `null !== $maxDepth`. 
About `count`, in general we try to avoid it as well for performance reasons. Could you increment a `counter` each time we visit a new path instead?